pkgs/applications/networking/instant-messengers/teams/default.nix

```diff
@@ -52,6 +52,23 @@ stdenv.mkDerivation rec {
     ln -s $out/opt/teams/teams $out/bin/
   '';
 
+  dontAutoPatchelf = true;
+
+  # Include runtimeDependencies in the RPATH of the included Node modules
```

Comment:
`autoPatchelfHook` also supports `runtimeDependencies` flag to include pulseaudio. Does this helps?

Reply:
https://github.com/NixOS/nixpkgs/blob/f3282c8d1e0ce6ba5d9f6aeddcfad51d879c7a4a/pkgs/applications/audio/bitwig-studio/bitwig-studio2.nix#L13

Reply:
`runtimeDependencies` is already [used by this derivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/teams/default.nix#L34).

Reply:
@Mic92 It was made this way before, and the call module was falling back on ALSA.

Reply:
To elaborate a bit: the libraries from `runtimeDependencies` are not propagated to the dynamically loadable node modules because of [this condition](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/auto-patchelf.sh#L105) in `autoPatchElfHook` because the `*.node` modules have `Type: DYN (Shared object file)` instead of `EXEC` or `INTERP` [it expects](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/auto-patchelf.sh#L18).

Reply:
Ok. Can add a comment regarding this?